pkg/agent/controller/networkpolicy/priority.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package networkpolicy
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"strconv"
+
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+)
+
+const (
+	PriorityBottomCNP     = uint16(100)
+	InitialPriorityOffest = uint16(130)
+	InitialPriorityZones  = 100
+	DefaultTierStart      = uint16(13100)
```

Comment:
Just feel this number looks strange. Could we use another one like 10,000 or 10K?

Reply:
We want to reserve some priorities and make use of the 65,000+ OF priorities as much as possible. 10,000 for a tier zone will leave a lot unused priorities.

Reply:
But it is just based on the estimate of 5 tiers at most, while in API we do not put any limit of the number of tiers? How if user creates more than 5 tiers?

Reply:
Right now this is not yet supported. We just wanted to divide the initial assumed five tier evenly to start with, which is also what @abhiraut had in mind initially. I could refactor the numbers to 100 + 100*100 = 10100 for default tier start. It would mean we can only support 100 unique priorities in a single tier zone at this point.

Reply:
We have not started to implement tiering yet right? And when we implement it, we do not put a limit of 5 tiers? Then what is the point of using the number of 64K/5?

Reply:
Yes and yes. It's the 'default division' for cases where user only has five or less tiers. From my understanding creation of more tiers would not be a case often, and we wanted each tier to be as 'dilated' as possible if there's only five tiers. This is merely a design choice and for the next release the algorithm can change arbitrarily without affecting upgrade.

Reply:
I do not get it. Do you mean when we implement tiering, for the first version we would limit the tiers to at most five? If not, when you have more than five, how you plan to implement that? Creating a new table or still using a single table? But with either way, you can not save any code complexity to put a limit of 64K/5 for flow priorities? Am I missing anything?

Reply:
Discussed offline and agreed on DefaultTierStart number will not have upgrade impacts. This can be 13,100 or 10,100 or 10,000. I am open to suggestions whether we want to keep this number or change it to something else